tensorflow/core/lib/io/buffered_inputstream.cc

```diff
@@ -156,15 +156,24 @@ Status BufferedInputStream::Seek(int64 position) {
                                    position);
   }
 
-  // Position of the buffer within file.
-  const int64 bufpos = Tell();
-  if (position < bufpos) {
-    // Reset input stream and skip 'position' bytes.
+  // Position of the buffer's lower limit within file.
+  const int64 buf_lower_limit  = input_stream_->Tell() - limit_ ;
```

Comment:
I don't think Tell() - limit_ is the right way to get the lower bound. I think it should be Tell() - pos_.

Reply:
@rohan100jain  i drew a diagram for better illustration, let me know if you still have any question.

https://www.dropbox.com/s/0z4fv7u7atroixh/Photo%20Oct%2012%2C%203%2025%2056%20PM.jpg?dl=0